### PR TITLE
Depth Estimation Transformers

### PR DESCRIPTION
**Pull request by jacobmarks**
Adds Monocular Depth Estimation transformers to Hugging Face integration.

- Apply HF `ForDepthEstimation` models directly to FiftyOne datasets
- Load HF depth estimation models from the FiftyOne Model Zoo
- Document changes to integration

```py
import fiftyone.zoo as foz

model = foz.load_zoo_model(
     "depth-estimation-transformer-torch",
     name_or_path="Intel/dpt-hybrid-midas"",
)

dataset.apply_model(model, label_field="dpt_hybrid_midas")

session = fo.launch_app(dataset)
```

(Please fill in changes proposed in this fix)

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [x] Other

### CODE REVIEW

File: docs/source/integrations/huggingface.rst [hunk 1]
```diff
@@ -372,6 +372,57 @@ model's name or path as a keyword argument:
...
+model's name or path as a keyword argument:
+
+.. code-block:: python
+    :linenos:
+
+    import fiftyone.zoo as foz
+
+    model = foz.load_zoo_model(
+        "depth-estimation-transformer-torch",
+        name_or_path="Intel/dpt-hybrid-midas"",
```
Comment (ehofesmann) [inline docs/source/integrations/huggingface.rst:418]:
```suggestion
        name_or_path="Intel/dpt-hybrid-midas",
```

**Review by ehofesmann — COMMENTED** [review]
Aside from the one small comment, this looks/works great! Will approve once the comment is resolved :+1:

**Comment by jacobmarks** [comment]
Thanks @ehofesmann! Change made :)

**Review by ehofesmann — DISMISSED** [review]
Nice! This LGTM!

**Review by brimoor — APPROVED** [review]
LGTM

**Comment by brimoor** [comment]
This PR is deprecated by https://github.com/voxel51/fiftyone/pull/4082.

**Pull request closed without merging**